src/odemis/dataio/tiff.py

```diff
@@ -1835,8 +1826,58 @@ def _readAndMergeImages(self):
             imset[tiff_info_item['hdim_index']] = image
 
         return model.DataArray(imset, metadata=self.metadata)
+    
+
+class DataArrayShadowPyramidalTIFF(DataArrayShadowTIFF):
+    """
+    This class implements the read of a TIFF file
+    It has all the useful attributes of a DataArray and the actual data. It also implements
+    the reading of a pyramidal TIFF file. IOW, reading subdirectories and tiles.
+    """
+
+    def __init__(self, tiff_info, shape, dtype, metadata=None):
+        """
+        Constructor
+        tiff_info (dictionary or list of dictionaries): Information about the source tiff file
+            and directory from which the image should be read. It can be a dictionary or
+            a list of dictionaries. It is a list of dictionaries when
+            the DataArray has multiple pixelData
+            The dictionary (or each dictionary in the list) has 2 values:
+            'tiff_file' (handle): Handle of the tiff file
+            'dir_index' (int): Index of the directory
+        shape (tuple of int): The shape of the corresponding DataArray
+        dtype (numpy.dtype): The data type
+        metadata (dict str->val): The metadata
+        maxzoom (0<=int): the maximum zoom level possible. If the data isn't
```

Comment:
Neither this class nor DataArrayShadowTIFF have maxzoom/tile_shape arguments. Please remove them from the docstrings.